### ISSUE TITLE
Using System.MathF in core math functions 

### ISSUE THREAD

**Issue by Jorenkv** [dotnet/fsharp#12009]
I noticed the F# compiler uses `System.Math` for both `float` and `float32` values. Could it use `System.MathF` for `float32` values instead, when `System.MathF` is available? This would allow for better codegen by avoiding conversions between `float` and `float32` and using faster hardware instructions.

Simple codegen comparison: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEAzAzmgFxAENcBbAHwgAcYA7AAgGUBPXAmMgWACheMYBBrgCWdAIIMAFAhANsGCMQIBmAEwBKBgF5hYhgn6C9dAELTZ8xcvVbdAWWUALAGIA6JmJkbeRobgBHKAJJGTkFJVVNHWEgoUMeAX848zCrSNsYxwJXDzjvXx4gA==

The obvious alternative would be to do nothing.

**Comment by Happypig375** [comment]
Since a .NET Standard 2.1 version of FSharp.Core now exists, this can be implemented in that version, but not .NET Standard 2.0.

**Comment by dsyme** [comment]
> Since a .NET Standard 2.1 version of FSharp.Core now exists, this can be implemented in that version, but not .NET Standard 2.0.

Yes, this should be done

**Comment by vzarytovskii** [comment]
> > Since a .NET Standard 2.1 version of FSharp.Core now exists, this can be implemented in that version, but not .NET Standard 2.0.
> 
> 
> 
> Yes, this should be done

Do we need an RFC for it? It's relatively small feature.

**Comment by dsyme** [comment]
Yes, it's best to have an RFC to discuss I suppose - there is a real compat issue here that the functions may give different results in some circumstances.

**Comment by vzarytovskii** [comment]
@Jorenkv you can submit an RFC PR if you'd like here: https://github.com/fsharp/fslang-design
